pkg/devfile/convert/convert.go

```
@@ -0,0 +1,284 @@
+	if err != nil {
+		return nil, "", err
+	}
+	imageStream, err := client.GetImageStream(imageNS, imageName, imageTag)
+	if err != nil {
+		return nil, "", err
+	}
+
+	imageStreamImage, err := client.GetImageStreamImage(imageStream, imageTag)
+	if err != nil {
+		return nil, "", err
+	}
+
+	imageforDevfile := imageStream.Spec.Tags[0].From.Name
+
+	return imageStreamImage, imageforDevfile, nil
+}
+
+// setDevfileCommandsForS2I sets command in devfile.yaml from s2i data.
+func setDevfileCommandsForS2I(d data.DevfileData) {
+	klog.V(2).Info("Set devfile commands from s2i data")
+
+	buildCommand := devfilev1.Command{
+		Id: buildCommandID,
+		CommandUnion: devfilev1.CommandUnion{
+			Exec: &devfilev1.ExecCommand{
+				Component:   containerName,
+				CommandLine: buildCommandS2i,
+				LabeledCommand: devfilev1.LabeledCommand{
+					BaseCommand: devfilev1.BaseCommand{
+						Group: &devfilev1.CommandGroup{
+							Kind:      devfilev1.BuildCommandGroupKind,
+							IsDefault: true,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	runCommand := devfilev1.Command{
+		Id: runCommandID,
+		CommandUnion: devfilev1.CommandUnion{
+			Exec: &devfilev1.ExecCommand{
+				Component:   containerName,
+				CommandLine: runCommandS2i,
+				LabeledCommand: devfilev1.LabeledCommand{
+					BaseCommand: devfilev1.BaseCommand{
+						Group: &devfilev1.CommandGroup{
+							Kind:      devfilev1.RunCommandGroupKind,
+							IsDefault: true,
+						},
+					},
+				},
+			},
+		},
+	}
+	// Ignoring error as we are writing new file
+	_ = d.AddCommands(buildCommand, runCommand)
+
+}
+
+// setDevfileComponentsForS2I sets the devfile.yaml components field from s2i data.
+func setDevfileComponentsForS2I(d data.DevfileData, s2iImage string, localConfig *config.LocalConfigInfo, s2iEnv config.EnvVarList) error {
+	klog.V(2).Info("Set devfile components from s2i data")
+
+	maxMemory := localConfig.GetMaxMemory()
+	volumes, _ := localConfig.ListStorage()
+	urls, _ := localConfig.ListURLs()
```

Why are we ignoring the errors here?